regnum2phyx/regnum2phyx.js

```diff
@@ -0,0 +1,287 @@
+  let type = citation.citation_type || 'article';
+
+  // Regnum uses 'journal' where BibJSON uses 'article'.
+  if (type === 'journal') type = 'article';
+
+  // In BibJSON, identifiers are kept separate from URLs.
+  const identifiers = [];
+  if (citation.doi) identifiers.push({ type: 'doi', id: citation.doi });
+
+  const urls = [];
+  if (citation.url) urls.push({ url: citation.url });
+
+  // lodash.pickBy will remove empty keys from the object.
+  const entry = pickBy({
+    type,
+    title: (citation.title || '').trim(),
+    section_title: (citation.section_title || '').trim(),
+    year: Number((citation.year || '')),
+    edition: (citation.edition || '').trim(),
+    authors: convertAuthorsIntoBibJSON(citation.authors),
+    editors: convertAuthorsIntoBibJSON(citation.editors || []),
+    series_editors: convertAuthorsIntoBibJSON(citation.series_editors || []),
+    publisher: (citation.publisher || '').trim(),
+    city: (citation.city || '').trim(),
+    pages: (citation.pages || '').trim(),
+    figure: (citation.figure || '').trim(),
+    // keywords: (citation.keyword || '').trim(), -- We really don't need this.
+    identifier: identifiers,
+    link: urls,
+  });
+
+  if (type === 'book' || type === 'book_section') {
+    // Add the ISBN to the entry itself.
+    if (citation.isbn) {
+      identifiers.push({ type: 'isbn', id: citation.isbn });
+    }
+  } else if (type === 'article') {
+    // Identify journal identifiers, such as the
+    const journalIdentifiers = [];
+    if (citation.isbn) {
+      journalIdentifiers.push({ type: 'isbn', id: citation.isbn });
+    }
+    if (citation.issn) {
+      journalIdentifiers.push({ type: 'issn', id: citation.issn });
+    }
+
+    // In BibJSON (http://okfnlabs.org/bibjson/), the journal name, volume, number
+    // and pages should go into the 'journal' object.
+    entry.journal = pickBy({
+      name: (citation.journal || '').trim(),
+      volume: (citation.volume || '').trim(),
+      number: (citation.number || '').trim(),
+      identifier: journalIdentifiers,
+    });
+
+    // Since we've moved pages and ISBN into journal, we don't also need it in the main entry.
+    if (has(entry, 'pages')) delete entry.pages;
+  } else {
+    process.stderr.write(`Unknown citation type: '${type}', using anyway.`);
+  }
+
+  return [entry];
+}
+
+// Command line application starts here.
+
+// Read command-line arguments.
+const argv = yargs
+  .usage('Usage: $0 <JSON file to process> -o <directory to write files to>')
+  .demandCommand(1, 1) // Make sure there's one and only one file to convert.
+  .option('output-dir', {
+    alias: 'o',
+    describe: 'Directory to write Phyx files to',
+  })
+  .demandOption(['o'])
+  .help('h')
+  .alias('h', 'help')
+  .argv;
+
+// Read the database dump.
+const dump = JSON.parse(fs.readFileSync(argv._[0], 'utf8'));
+
+// This dump consists of multiple named phyloreferences, each of which should be
```

Comment:
You mean phylogenetic clade definition records, right? Regnum doesn't store Phyloreferences in the strict sense (which we should be using) but phylogenetic clade definitions in semi-structured records.